kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/manager/ResourceManager.java

```diff
@@ -110,6 +130,9 @@ private final <T extends HasMetadata> void createResource(boolean waitReady, T..
 
             assert type != null;
             type.create(resource);
+
+            storedResources.computeIfAbsent(getTestContext().getDisplayName(), k -> new Stack<>())
```

Comment:
displayName isn't a good choice of key.  Two tests could have the same display name.  I think getTestContext().getUniqueId() appears to be a better choice.

Reply:
UniqueId is a little bit ugly to log it, so I'll compose the `class name + displayName` that is unique as well as two tests cannot have the same name in the same class.

Reply:
The developer is free to use the @DisplayName annotation any way they please.    There's no guarantee of uniqueness. 

> UniqueId is a little bit ugly to log it

There's nothing to stop you using the UniqueId as the key and logging something else.